Longest substring ch20

Yuvraj-kadale opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.
You can learn more about contributing to NeoAlgo here: https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md

Happy Contributing!

-->

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
YES


### Description

Added longest substring without repeating characters in C, having a linear time complexity [ O(n) ] and constant space Complexity O(1). 

||Output||
![Capture](https://user-images.githubusercontent.com/43489758/92156016-f2b00f00-ee45-11ea-9969-615e46aee3af.JPG)

--Some other input-output---
![Capture](https://user-images.githubusercontent.com/43489758/92264846-7e847280-eefc-11ea-94cd-fde382804eb2.JPG)

![Capture](https://user-images.githubusercontent.com/43489758/92265292-e5099080-eefc-11ea-854b-ae9139b27955.JPG)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

issue #563 Solved !!

Yuvraj-kadale committed:
Added Longest substring without repeating Chars

Given a string, the program find the logest substring possible without repeating characters.

Yuvraj-kadale committed:
updated Readme

Added longest substring without repeating characters in C language.

iamrajiv reviewed (changes requested):
@Yuvraj-kadale add sample input-output, time complexity and space complexity

iamrajiv reviewed (changes requested):
@Yuvraj-kadale add sample input-output, time complexity and space complexity in `Longest substring without repeating chars.cpp` program in the form of comments. Also, move this program to `C/cp` not in `C/other`.

Abhijit2505 reviewed (changes requested):
Your code is good, but you need to place your code in the proper place as reviewed by @iamrajiv .

Yuvraj-kadale committed:
Update Longest substring without repeating chars.cpp

Abhijit2505 commented:
Please work upon the merge conflict thing, make a pull from the master repo to the forked repo.

Yuvraj-kadale committed:
Update Longest substring without repeating chars.cpp

Yuvraj-kadale committed:
Merge pull request #1 from Yuvraj-kadale/Longest-substring-CH20

Longest substring ch20

Yuvraj-kadale committed:
Add files via upload

Yuvraj-kadale committed:
Add files via upload

Yuvraj-kadale committed:
Add files via upload

Yuvraj-kadale committed:
Add files via upload

Yuvraj-kadale committed:
Merge pull request #2 from Yuvraj-kadale/Yuvraj-kadale-patch-1-1

Yuvraj kadale patch 1 1

iamrajiv reviewed (changes requested):
@Yuvraj-kadale Do not edit `.all-contributorsrc` and `CONTRIBUTORS.md ` files. These files are managed by All Contributors bot.

iamrajiv commented on C/other/Longest substring without repeating chars.cpp:
```diff
@@ -0,0 +1,77 @@
+
+//Input: 
+//Output: length of longest substring without repeating charecters: 0
+------------------------------------------------------------------------------
```
@Yuvraj-kadale Remove these type of line patterns.

Yuvraj-kadale replied:
Done removed them!!

iamrajiv commented on C/other/Longest substring without repeating chars.cpp:
```diff
@@ -0,0 +1,77 @@
+//This code is submitted to NeoAlgo as a part of Contributors hack 2020
```
@Yuvraj-kadale Make all single-line comments in one multiline comment. Something like this
```
/*
...
*/
```

Yuvraj-kadale replied:
Done!!

Yuvraj-kadale committed:
location updated 

Changed the file location to C/cp from C/others and updated the comments in the code

Yuvraj-kadale committed:
Delete .all-contributorsrc

Yuvraj-kadale commented:
> 
> 
> @Yuvraj-kadale add sample input-output, time complexity and space complexity in `Longest substring without repeating chars.cpp` program in the form of comments. Also, move this program to `C/cp` not in `C/other`.

Done, file has been moved to C/cp from C/others

Yuvraj-kadale commented:
> 
> 
> Please work upon the merge conflict thing, make a pull from the master repo to the forked repo.

I just saw in the PR the merge conflicts are due to updation in the readme files and contributors files!
https://github.com/TesseractCoding/NeoAlgo/pull/902#pullrequestreview-486961867

should I send a new PR!? and close this PR!?

atarax665 commented:
@Yuvraj-kadale I am closing this PR as it has so many conflicts, open a new PR for this issue.
